### PR TITLE
[BLOG] Recreate Diane's 'integrating Hypothesis into SymPy' blog post

### PR DESCRIPTION
**Pull request by bskinn**
Recreation of #773 from bbfa665. Attn @melissawm.

## Text styling

- [ ] The blog is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags in order to do so (with only one level-one header).
- [ ] All links describe where they link to (for example, check the [Quansight labs website](https://labs.quansight.org/)).
- [ ] Any kind of styling that the author uses (for example, bold for emphasis) is consistent throughout the blog.

## Non-text contents

- [ ] Blog post featured image is in PNG or JPEG format, **not** SVG.
- [ ] All content is represented as text (for example, images need alt text and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If it were to be read as plain text, the blog still makes sense and no information is missing.

### CODE REVIEW

**Comment by melissawm** [comment]
@asmeurer and @honno if you are happy with this post, please approve and we can merge and have it published soon. Thanks!

File: apps/labs/posts/integrating-hypothesis-into-sympy.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    assert h.degree() == f.degree() + g.degree()
+```
+
+Note, that here `polys()` is custom-built for sympy and generates a random polynomial with integer coefficients. It is not a built-in Hypothesis strategy.
+
+### How Hypothesis Works
+
+Give Hypothesis the types of inputs you are expecting using the _@given_ decorator and it will automatically generate examples using the _strategies_ module. It will then use these examples to test and report the minimal failing inputs (if any).
+
+Hypothesis is able to come up with interesting inputs using a combination of smart generation, guiding metrics, and feedback loops. For example, the built-in stategies like `integers()` have default behaviors tuned for common useful values. That is, while some inputs are random, it also tries to choose cases that commonly cause errors (like 0 or NaN). Hypothesis is _adversarial_ in the way it chooses inputs to test against your function.
```
Comment (honno) [inline apps/labs/posts/integrating-hypothesis-into-sympy.md:73]:
```suggestion
Hypothesis is able to come up with interesting inputs using a combination of smart generation, guiding metrics, and feedback loops. For example, the built-in strategies like `floats()` have default behaviors tuned for common useful values. That is, while some inputs are random, it also tries to choose cases that commonly cause errors (like 0 or NaN). Hypothesis is _adversarial_ in the way it chooses inputs to test against your function.
```

1. Grammar fix
2. `floats()` is the better example, prob what Diane was going for.

File: apps/labs/posts/integrating-hypothesis-into-sympy.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+Hypothesis comes with built-in strategy functions for common Python data types. In the example above, we accessed integers using `st.integers()`, but Hypothesis also gives you access to `floats()`, `booleans()`, `fractions()`, `dictionaries()`, and many more.
+
+The full documentation for Hypothesis can be found [here](https://hypothesis.readthedocs.io/en/latest/index.html) and for a nice and robust introduction, check out this video from [PyCon 2019](https://youtu.be/KcyGUVzL7HA?si=lglSRamsWsY1YLIR).
+
+Overall, Hypothesis is great at finding bugs and in general, writing tests as high level properties keeps your code consistent.
+
+### What Hypothesis is NOT
+
+- Hypothesis cannot test black boxes, machine learning systems, simulations of complicated systems, or code with lots of state (e.g., things which depend on a database or talk to a network etc.). Hypothesis must receive an understanding of the input and output behavior that can be easily modelled.
```
Comment (honno) [inline apps/labs/posts/integrating-hypothesis-into-sympy.md:83]:
```suggestion
- It's tricky to test black boxes, machine learning systems, simulations of complicated systems, or code with lots of state (e.g., things which depend on a database or talk to a network etc.) with Hypothesis. Hypothesis must receive an understanding of the input and output behavior that can be easily modelled.
```

**Review by honno — APPROVED** [review]
LGTM! Only some nit picks. Thanks for the blog post @dianetc and Brian/Melissa/Aaron for reviews.

**Pull request merged**